### PR TITLE
[Feature] Support DDRNet

### PR DESCRIPTION
**Pull request by whu-pzhang**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support DDRNet
Paper: [Deep Dual-resolution Networks for Real-time and Accurate Semantic Segmentation of Road Scenes](https://arxiv.org/pdf/2101.06085)
official Code: https://github.com/ydhongHIT/DDRNet

There is already a PR https://github.com/open-mmlab/mmsegmentation/pull/1722 , but it has been inactive for a long time.

#### inference with converted official weights

| Method | Backbone      | mIoU(official) | mIoU(converted weight) |
| ------ | ------------- | -------------- | ---------------------- |
| DDRNet | DDRNet23-slim | 77.8           | 77.84                  |
| DDRNet | DDRNet23      | 79.5           | 79.53                  |     

#### training with converted pretrained backbone

| Method | Backbone      | Crop Size | Lr schd  | Inf time(fps) | Device   | mIoU  | mIoU(ms+flip) | config       | download     |
| ------ | ------------- | --------- | ------- | ------- | -------- | ----- | ------------- | ------------ | ------------ |
| DDRNet | DDRNet23-slim | 1024x1024 | 120000   | 85.85         | RTX 8000 | 77.85 | 79.80         | [config](https://github.com/whu-pzhang/mmsegmentation/blob/ddrnet/configs/ddrnet/ddrnet_23-slim_in1k-pre_2xb6-120k_cityscapes-1024x1024.py) | model \| log |
| DDRNet | DDRNet23      | 1024x1024 | 120000     | 33.41         | RTX 8000 | 79.53 | 80.98         | [config](https://github.com/whu-pzhang/mmsegmentation/blob/ddrnet/configs/ddrnet/ddrnet_23_in1k-pre_2xb6-120k_cityscapes-1024x1024.py) | model \| log |

The converted pretrained backbone weights download link:

1. [ddrnet23s_in1k_mmseg.pth](https://drive.google.com/file/d/1Ni4F1PMGGjuld-1S9fzDTmneLfpMuPTG/view?usp=sharing)
2. [ddrnet23_in1k_mmseg.pth](https://drive.google.com/file/d/11rsijC1xOWB6B0LgNQkAG-W6e1OdbCyJ/view?usp=sharing)

## To do

- [x] support inference with converted official weights
- [x] support training on cityscapes dataset

### CODE REVIEW

**Comment by kaizhong2021** [comment]
Hi, how to download mmseg style checkpoint file?
I ran into an error like this using official pretrained weights
unexpected key in source state_dict: model.conv1.0.weight, model.conv1.0.bias, model.conv1.1.weight, model.conv1.1.bias, model.conv1.1.running_mean, model.conv1.1.running_var, model.conv1.1.num_batches_tracked, model.conv1.3.weight, model.conv1.3.bias, model.conv1.4.weight, model.conv1.4.bias, model.conv1.4.running_mean, model.conv1.4.running_var, model.conv1.4.num_batches_tracked, model.layer1.0.conv1.weight, model.layer1.0.bn1.weight, model.layer1.0.bn1.bias, model.layer1.0.bn1.running_mean, model.layer1.0.bn1.running_var, model.layer1.0.bn1.num_batches_tracked, model.layer1.0.conv2.weight, model.layer1.0.bn2.weight, model.layer1.0.bn2.bias, model.layer1.0.bn2.running_mean, model.layer1.0.bn2.running_var, model.layer1.0.bn2.num_batches_tracked, model.layer1.1.conv1.weight, model.layer1.1.bn1.weight, model.layer1.1.bn1.bias, model.layer1.1.bn1.running_mean, model.layer1.1.bn1.running_var, model.layer1.1.bn1.num_batches_tracked, model.layer1.1.conv2.weight, model.layer1.1.bn2.weight, model.layer1.1.bn2.bias, model.layer1.1.bn2.running_mean, model.layer1.1.bn2.running_var, model.layer1.1.bn2.num_batches_tracked, model.layer2.0.conv1.weight, model.layer2.0.bn1.weight, model.layer2.0.bn1.bias, model.layer2.0.bn1.running_mean, model.layer2.0.bn1.running_var, model.layer2.0.bn1.num_batches_tracked, model.layer2.0.conv2.weight, model.layer2.0.bn2.weight, model.layer2.0.bn2.bias, model.layer2.0.bn2.running_mean, model.layer2.0.bn2.running_var, model.layer2.0.bn2.num_batches_tracked, model.layer2.0.downsample.0.weight, model.layer2.0.downsample.1.weight, model.layer2.0.downsample.1.bias, model.layer2.0.downsample.1.running_mean, model.layer2.0.downsample.1.running_var, model.layer2.0.downsample.1.num_batches_tracked, model.layer2.1.conv1.weight, model.layer2.1.bn1.weight, model.layer2.1.bn1.bias, model.layer2.1.bn1.running_mean, model.layer2.1.bn1.running_var, model.layer2.1.bn1.num_batches_tracked, model.layer2.1.conv2.weight, model.layer2.1.bn2.weight, model.layer2.1.bn2.bias, model.layer2.1.bn2.running_mean, model.layer2.1.bn2.running_var, model.layer2.1.bn2.num_batches_tracked, model.layer3.0.conv1.weight, model.layer3.0.bn1.weight, model.layer3.0.bn1.bias, model.layer3.0.bn1.running_mean, model.layer3.0.bn1.running_var, model.layer3.0.bn1.num_batches_tracked, model.layer3.0.conv2.weight, model.layer3.0.bn2.weight, model.layer3.0.bn2.bias, model.layer3.0.bn2.running_mean, model.layer3.0.bn2.running_var, model.layer3.0.bn2.num_batches_tracked, model.layer3.0.downsample.0.weight, model.layer3.0.downsample.1.weight, model.layer3.0.downsample.1.bias, model.layer3.0.downsample.1.running_mean, model.layer3.0.downsample.1.running_var, model.layer3.0.downsample.1.num_batches_tracked, model.layer3.1.conv1.weight, model.layer3.1.bn1.weight, model.layer3.1.bn1.bias, model.layer3.1.bn1.running_mean, model.layer3.1.bn1.running_var, model.layer3.1.bn1.num_batches_tracked, model.layer3.1.conv2.weight, model.layer3.1.bn2.weight, model.layer3.1.bn2.bias, model.layer3.1.bn2.running_mean, model.layer3.1.bn2.running_var, model.layer3.1.bn2.num_batches_tracked, model.layer4.0.conv1.weight, model.layer4.0.bn1.weight, model.layer4.0.bn1.bias, model.layer4.0.bn1.running_mean, model.layer4.0.bn1.running_var, model.layer4.0.bn1.num_batches_tracked, model.layer4.0.conv2.weight, model.layer4.0.bn2.weight, model.layer4.0.bn2.bias, model.layer4.0.bn2.running_mean, model.layer4.0.bn2.running_var, model.layer4.0.bn2.num_batches_tracked, model.layer4.0.downsample.0.weight, model.layer4.0.downsample.1.weight, model.layer4.0.downsample.1.bias, model.layer4.0.downsample.1.running_mean, model.layer4.0.downsample.1.running_var, model.layer4.0.downsample.1.num_batches_tracked, model.layer4.1.conv1.weight, model.layer4.1.bn1.weight, model.layer4.1.bn1.bias, model.layer4.1.bn1.running_mean, model.layer4.1.bn1.running_var, model.layer4.1.bn1.num_batches_tracked, model.layer4.1.conv2.weight, model.layer4.1.bn2.weight, model.layer4.1.bn2.bias, model.layer4.1.bn2.running_mean, model.layer4.1.bn2.running_var, model.layer4.1.bn2.num_batches_tracked, model.compression3.0.weight, model.compression3.1.weight, model.compression3.1.bias, model.compression3.1.running_mean, model.compression3.1.running_var, model.compression3.1.num_batches_tracked, model.compression4.0.weight, model.compression4.1.weight, model.compression4.1.bias, model.compression4.1.running_mean, model.compression4.1.running_var, model.compression4.1.num_batches_tracked, model.down3.0.weight, model.down3.1.weight, model.down3.1.bias, model.down3.1.running_mean, model.down3.1.running_var, model.down3.1.num_batches_tracked, model.down4.0.weight, model.down4.1.weight, model.down4.1.bias, model.down4.1.running_mean, model.down4.1.running_var, model.down4.1.num_batches_tracked, model.down4.3.weight, model.down4.4.weight, model.down4.4.bias, model.down4.4.running_mean, model.down4.4.running_var, model.down4.4.num_batches_tracked, model.layer3_.0.conv1.weight, model.layer3_.0.bn1.weight, model.layer3_.0.bn1.bias, model.layer3_.0.bn1.running_mean, model.layer3_.0.bn1.running_var, model.layer3_.0.bn1.num_batches_tracked, model.layer3_.0.conv2.weight, model.layer3_.0.bn2.weight, model.layer3_.0.bn2.bias, model.layer3_.0.bn2.running_mean, model.layer3_.0.bn2.running_var, model.layer3_.0.bn2.num_batches_tracked, model.layer3_.1.conv1.weight, model.layer3_.1.bn1.weight, model.layer3_.1.bn1.bias, model.layer3_.1.bn1.running_mean, model.layer3_.1.bn1.running_var, model.layer3_.1.bn1.num_batches_tracked, model.layer3_.1.conv2.weight, model.layer3_.1.bn2.weight, model.layer3_.1.bn2.bias, model.layer3_.1.bn2.running_mean, model.layer3_.1.bn2.running_var, model.layer3_.1.bn2.num_batches_tracked, model.layer4_.0.conv1.weight, model.layer4_.0.bn1.weight, model.layer4_.0.bn1.bias, model.layer4_.0.bn1.running_mean, model.layer4_.0.bn1.running_var, model.layer4_.0.bn1.num_batches_tracked, model.layer4_.0.conv2.weight, model.layer4_.0.bn2.weight, model.layer4_.0.bn2.bias, model.layer4_.0.bn2.running_mean, model.layer4_.0.bn2.running_var, model.layer4_.0.bn2.num_batches_tracked, model.layer4_.1.conv1.weight, model.layer4_.1.bn1.weight, model.layer4_.1.bn1.bias, model.layer4_.1.bn1.running_mean, model.layer4_.1.bn1.running_var, model.layer4_.1.bn1.num_batches_tracked, model.layer4_.1.conv2.weight, model.layer4_.1.bn2.weight, model.layer4_.1.bn2.bias, model.layer4_.1.bn2.running_mean, model.layer4_.1.bn2.running_var, model.layer4_.1.bn2.num_batches_tracked, model.layer5_.0.conv1.weight, model.layer5_.0.bn1.weight, model.layer5_.0.bn1.bias, model.layer5_.0.bn1.running_mean, model.layer5_.0.bn1.running_var, model.layer5_.0.bn1.num_batches_tracked, model.layer5_.0.conv2.weight, model.layer5_.0.bn2.weight, model.layer5_.0.bn2.bias, model.layer5_.0.bn2.running_mean, model.layer5_.0.bn2.running_var, model.layer5_.0.bn2.num_batches_tracked, model.layer5_.0.conv3.weight, model.layer5_.0.bn3.weight, model.layer5_.0.bn3.bias, model.layer5_.0.bn3.running_mean, model.layer5_.0.bn3.running_var, model.layer5_.0.bn3.num_batches_tracked, model.layer5_.0.downsample.0.weight, model.layer5_.0.downsample.1.weight, model.layer5_.0.downsample.1.bias, model.layer5_.0.downsample.1.running_mean, model.layer5_.0.downsample.1.running_var, model.layer5_.0.downsample.1.num_batches_tracked, model.layer5.0.conv1.weight, model.layer5.0.bn1.weight, model.layer5.0.bn1.bias, model.layer5.0.bn1.running_mean, model.layer5.0.bn1.running_var, model.layer5.0.bn1.num_batches_tracked, model.layer5.0.conv2.weight, model.layer5.0.bn2.weight, model.layer5.0.bn2.bias, model.layer5.0.bn2.running_mean, model.layer5.0.bn2.running_var, model.layer5.0.bn2.num_batches_tracked, model.layer5.0.conv3.weight, model.layer5.0.bn3.weight, model.layer5.0.bn3.bias, model.layer5.0.bn3.running_mean, model.layer5.0.bn3.running_var, model.layer5.0.bn3.num_batches_tracked, model.layer5.0.downsample.0.weight, model.layer5.0.downsample.1.weight, model.layer5.0.downsample.1.bias, model.layer5.0.downsample.1.running_mean, model.layer5.0.downsample.1.running_var, model.layer5.0.downsample.1.num_batches_tracked, model.spp.scale1.1.weight, model.spp.scale1.1.bias, model.spp.scale1.1.running_mean, model.spp.scale1.1.running_var, model.spp.scale1.1.num_batches_tracked, model.spp.scale1.3.weight, model.spp.scale2.1.weight, model.spp.scale2.1.bias, model.spp.scale2.1.running_mean, model.spp.scale2.1.running_var, model.spp.scale2.1.num_batches_tracked, model.spp.scale2.3.weight, model.spp.scale3.1.weight, model.spp.scale3.1.bias, model.spp.scale3.1.running_mean, model.spp.scale3.1.running_var, model.spp.scale3.1.num_batches_tracked, model.spp.scale3.3.weight, model.spp.scale4.1.weight, model.spp.scale4.1.bias, model.spp.scale4.1.running_mean, model.spp.scale4.1.running_var, model.spp.scale4.1.num_batches_tracked, model.spp.scale4.3.weight, model.spp.scale0.0.weight, model.spp.scale0.0.bias, model.spp.scale0.0.running_mean, model.spp.scale0.0.running_var, model.spp.scale0.0.num_batches_tracked, model.spp.scale0.2.weight, model.spp.process1.0.weight, model.spp.process1.0.bias, model.spp.process1.0.running_mean, model.spp.process1.0.running_var, model.spp.process1.0.num_batches_tracked, model.spp.process1.2.weight, model.spp.process2.0.weight, model.spp.process2.0.bias, model.spp.process2.0.running_mean, model.spp.process2.0.running_var, model.spp.process2.0.num_batches_tracked, model.spp.process2.2.weight, model.spp.process3.0.weight, model.spp.process3.0.bias, model.spp.process3.0.running_mean, model.spp.process3.0.running_var, model.spp.process3.0.num_batches_tracked, model.spp.process3.2.weight, model.spp.process4.0.weight, model.spp.process4.0.bias, model.spp.process4.0.running_mean, model.spp.process4.0.running_var, model.spp.process4.0.num_batches_tracked, model.spp.process4.2.weight, model.spp.compression.0.weight, model.spp.compression.0.bias, model.spp.compression.0.running_mean, model.spp.compression.0.running_var, model.spp.compression.0.num_batches_tracked, model.spp.compression.2.weight, model.spp.shortcut.0.weight, model.spp.shortcut.0.bias, model.spp.shortcut.0.running_mean, model.spp.shortcut.0.running_var, model.spp.shortcut.0.num_batches_tracked, model.spp.shortcut.2.weight, model.seghead_extra.bn1.weight, model.seghead_extra.bn1.bias, model.seghead_extra.bn1.running_mean, model.seghead_extra.bn1.running_var, model.seghead_extra.bn1.num_batches_tracked, model.seghead_extra.conv1.weight, model.seghead_extra.bn2.weight, model.seghead_extra.bn2.bias, model.seghead_extra.bn2.running_mean, model.seghead_extra.bn2.running_var, model.seghead_extra.bn2.num_batches_tracked, model.seghead_extra.conv2.weight, model.seghead_extra.conv2.bias, model.final_layer.bn1.weight, model.final_layer.bn1.bias, model.final_layer.bn1.running_mean, model.final_layer.bn1.running_var, model.final_layer.bn1.num_batches_tracked, model.final_layer.conv1.weight, model.final_layer.bn2.weight, model.final_layer.bn2.bias, model.final_layer.bn2.running_mean, model.final_layer.bn2.running_var, model.final_layer.bn2.num_batches_tracked, model.final_layer.conv2.weight, model.final_layer.conv2.bias, loss.criterion.weight

**Comment by MengzhangLI** [comment]
The results look pretty good, the old pr was not merged mainly because I can not re-implement training metric. And it was not in developing schedule so I did not have time to figure it out.

**Comment by whu-pzhang** [comment]
> Hi, how to download mmseg style checkpoint file? I ran into an error like this using official pretrained weights unexpected key in source state_dict: model.conv1.0.weight, model.conv1.0.bias, model.conv1.1.weight, model.conv1.1.bias, model.conv1.1.running_mean, model.conv1.1.running_var, model.conv1.1.num_batches_tracked, model.conv1.3.weight, model.conv1.3.bias, model.conv1.4.weight, model.conv1.4.bias, model.conv1.4.running_mean, model.conv1.4.running_var, model.conv1.4.num_batches_tracked, model.layer1.0.conv1.weight, model.layer1.0.bn1.weight, model.layer1.0.bn1.bias, model.layer1.0.bn1.running_mean, model.layer1.0.bn1.running_var, model.layer1.0.bn1.num_batches_tracked, model.layer1.0.conv2.weight, model.layer1.0.bn2.weight, model.layer1.0.bn2.bias, model.layer1.0.bn2.running_mean, model.layer1.0.bn2.running_var, model.layer1.0.bn2.num_batches_tracked, model.layer1.1.conv1.weight, model.layer1.1.bn1.weight, model.layer1.1.bn1.bias, model.layer1.1.bn1.running_mean, model.layer1.1.bn1.running_var, model.layer1.1.bn1.num_batches_tracked, model.layer1.1.conv2.weight, model.layer1.1.bn2.weight, model.layer1.1.bn2.bias, model.layer1.1.bn2.running_mean, model.layer1.1.bn2.running_var, model.layer1.1.bn2.num_batches_tracked, model.layer2.0.conv1.weight, model.layer2.0.bn1.weight, model.layer2.0.bn1.bias, model.layer2.0.bn1.running_mean, model.layer2.0.bn1.running_var, model.layer2.0.bn1.num_batches_tracked, model.layer2.0.conv2.weight, model.layer2.0.bn2.weight, model.layer2.0.bn2.bias, model.layer2.0.bn2.running_mean, model.layer2.0.bn2.running_var, model.layer2.0.bn2.num_batches_tracked, model.layer2.0.downsample.0.weight, model.layer2.0.downsample.1.weight, model.layer2.0.downsample.1.bias, model.layer2.0.downsample.1.running_mean, model.layer2.0.downsample.1.running_var, model.layer2.0.downsample.1.num_batches_tracked, model.layer2.1.conv1.weight, model.layer2.1.bn1.weight, model.layer2.1.bn1.bias, model.layer2.1.bn1.running_mean, model.layer2.1.bn1.running_var, model.layer2.1.bn1.num_batches_tracked, model.layer2.1.conv2.weight, model.layer2.1.bn2.weight, model.layer2.1.bn2.bias, model.layer2.1.bn2.running_mean, model.layer2.1.bn2.running_var, model.layer2.1.bn2.num_batches_tracked, model.layer3.0.conv1.weight, model.layer3.0.bn1.weight, model.layer3.0.bn1.bias, model.layer3.0.bn1.running_mean, model.layer3.0.bn1.running_var, model.layer3.0.bn1.num_batches_tracked, model.layer3.0.conv2.weight, model.layer3.0.bn2.weight, model.layer3.0.bn2.bias, model.layer3.0.bn2.running_mean, model.layer3.0.bn2.running_var, model.layer3.0.bn2.num_batches_tracked, model.layer3.0.downsample.0.weight, model.layer3.0.downsample.1.weight, model.layer3.0.downsample.1.bias, model.layer3.0.downsample.1.running_mean, model.layer3.0.downsample.1.running_var, model.layer3.0.downsample.1.num_batches_tracked, model.layer3.1.conv1.weight, model.layer3.1.bn1.weight, model.layer3.1.bn1.bias, model.layer3.1.bn1.running_mean, model.layer3.1.bn1.running_var, model.layer3.1.bn1.num_batches_tracked, model.layer3.1.conv2.weight, model.layer3.1.bn2.weight, model.layer3.1.bn2.bias, model.layer3.1.bn2.running_mean, model.layer3.1.bn2.running_var, model.layer3.1.bn2.num_batches_tracked, model.layer4.0.conv1.weight, model.layer4.0.bn1.weight, model.layer4.0.bn1.bias, model.layer4.0.bn1.running_mean, model.layer4.0.bn1.running_var, model.layer4.0.bn1.num_batches_tracked, model.layer4.0.conv2.weight, model.layer4.0.bn2.weight, model.layer4.0.bn2.bias, model.layer4.0.bn2.running_mean, model.layer4.0.bn2.running_var, model.layer4.0.bn2.num_batches_tracked, model.layer4.0.downsample.0.weight, model.layer4.0.downsample.1.weight, model.layer4.0.downsample.1.bias, model.layer4.0.downsample.1.running_mean, model.layer4.0.downsample.1.running_var, model.layer4.0.downsample.1.num_batches_tracked, model.layer4.1.conv1.weight, model.layer4.1.bn1.weight, model.layer4.1.bn1.bias, model.layer4.1.bn1.running_mean, model.layer4.1.bn1.running_var, model.layer4.1.bn1.num_batches_tracked, model.layer4.1.conv2.weight, model.layer4.1.bn2.weight, model.layer4.1.bn2.bias, model.layer4.1.bn2.running_mean, model.layer4.1.bn2.running_var, model.layer4.1.bn2.num_batches_tracked, model.compression3.0.weight, model.compression3.1.weight, model.compression3.1.bias, model.compression3.1.running_mean, model.compression3.1.running_var, model.compression3.1.num_batches_tracked, model.compression4.0.weight, model.compression4.1.weight, model.compression4.1.bias, model.compression4.1.running_mean, model.compression4.1.running_var, model.compression4.1.num_batches_tracked, model.down3.0.weight, model.down3.1.weight, model.down3.1.bias, model.down3.1.running_mean, model.down3.1.running_var, model.down3.1.num_batches_tracked, model.down4.0.weight, model.down4.1.weight, model.down4.1.bias, model.down4.1.running_mean, model.down4.1.running_var, model.down4.1.num_batches_tracked, model.down4.3.weight, model.down4.4.weight, model.down4.4.bias, model.down4.4.running_mean, model.down4.4.running_var, model.down4.4.num_batches_tracked, model.layer3_.0.conv1.weight, model.layer3_.0.bn1.weight, model.layer3_.0.bn1.bias, model.layer3_.0.bn1.running_mean, model.layer3_.0.bn1.running_var, model.layer3_.0.bn1.num_batches_tracked, model.layer3_.0.conv2.weight, model.layer3_.0.bn2.weight, model.layer3_.0.bn2.bias, model.layer3_.0.bn2.running_mean, model.layer3_.0.bn2.running_var, model.layer3_.0.bn2.num_batches_tracked, model.layer3_.1.conv1.weight, model.layer3_.1.bn1.weight, model.layer3_.1.bn1.bias, model.layer3_.1.bn1.running_mean, model.layer3_.1.bn1.running_var, model.layer3_.1.bn1.num_batches_tracked, model.layer3_.1.conv2.weight, model.layer3_.1.bn2.weight, model.layer3_.1.bn2.bias, model.layer3_.1.bn2.running_mean, model.layer3_.1.bn2.running_var, model.layer3_.1.bn2.num_batches_tracked, model.layer4_.0.conv1.weight, model.layer4_.0.bn1.weight, model.layer4_.0.bn1.bias, model.layer4_.0.bn1.running_mean, model.layer4_.0.bn1.running_var, model.layer4_.0.bn1.num_batches_tracked, model.layer4_.0.conv2.weight, model.layer4_.0.bn2.weight, model.layer4_.0.bn2.bias, model.layer4_.0.bn2.running_mean, model.layer4_.0.bn2.running_var, model.layer4_.0.bn2.num_batches_tracked, model.layer4_.1.conv1.weight, model.layer4_.1.bn1.weight, model.layer4_.1.bn1.bias, model.layer4_.1.bn1.running_mean, model.layer4_.1.bn1.running_var, model.layer4_.1.bn1.num_batches_tracked, model.layer4_.1.conv2.weight, model.layer4_.1.bn2.weight, model.layer4_.1.bn2.bias, model.layer4_.1.bn2.running_mean, model.layer4_.1.bn2.running_var, model.layer4_.1.bn2.num_batches_tracked, model.layer5_.0.conv1.weight, model.layer5_.0.bn1.weight, model.layer5_.0.bn1.bias, model.layer5_.0.bn1.running_mean, model.layer5_.0.bn1.running_var, model.layer5_.0.bn1.num_batches_tracked, model.layer5_.0.conv2.weight, model.layer5_.0.bn2.weight, model.layer5_.0.bn2.bias, model.layer5_.0.bn2.running_mean, model.layer5_.0.bn2.running_var, model.layer5_.0.bn2.num_batches_tracked, model.layer5_.0.conv3.weight, model.layer5_.0.bn3.weight, model.layer5_.0.bn3.bias, model.layer5_.0.bn3.running_mean, model.layer5_.0.bn3.running_var, model.layer5_.0.bn3.num_batches_tracked, model.layer5_.0.downsample.0.weight, model.layer5_.0.downsample.1.weight, model.layer5_.0.downsample.1.bias, model.layer5_.0.downsample.1.running_mean, model.layer5_.0.downsample.1.running_var, model.layer5_.0.downsample.1.num_batches_tracked, model.layer5.0.conv1.weight, model.layer5.0.bn1.weight, model.layer5.0.bn1.bias, model.layer5.0.bn1.running_mean, model.layer5.0.bn1.running_var, model.layer5.0.bn1.num_batches_tracked, model.layer5.0.conv2.weight, model.layer5.0.bn2.weight, model.layer5.0.bn2.bias, model.layer5.0.bn2.running_mean, model.layer5.0.bn2.running_var, model.layer5.0.bn2.num_batches_tracked, model.layer5.0.conv3.weight, model.layer5.0.bn3.weight, model.layer5.0.bn3.bias, model.layer5.0.bn3.running_mean, model.layer5.0.bn3.running_var, model.layer5.0.bn3.num_batches_tracked, model.layer5.0.downsample.0.weight, model.layer5.0.downsample.1.weight, model.layer5.0.downsample.1.bias, model.layer5.0.downsample.1.running_mean, model.layer5.0.downsample.1.running_var, model.layer5.0.downsample.1.num_batches_tracked, model.spp.scale1.1.weight, model.spp.scale1.1.bias, model.spp.scale1.1.running_mean, model.spp.scale1.1.running_var, model.spp.scale1.1.num_batches_tracked, model.spp.scale1.3.weight, model.spp.scale2.1.weight, model.spp.scale2.1.bias, model.spp.scale2.1.running_mean, model.spp.scale2.1.running_var, model.spp.scale2.1.num_batches_tracked, model.spp.scale2.3.weight, model.spp.scale3.1.weight, model.spp.scale3.1.bias, model.spp.scale3.1.running_mean, model.spp.scale3.1.running_var, model.spp.scale3.1.num_batches_tracked, model.spp.scale3.3.weight, model.spp.scale4.1.weight, model.spp.scale4.1.bias, model.spp.scale4.1.running_mean, model.spp.scale4.1.running_var, model.spp.scale4.1.num_batches_tracked, model.spp.scale4.3.weight, model.spp.scale0.0.weight, model.spp.scale0.0.bias, model.spp.scale0.0.running_mean, model.spp.scale0.0.running_var, model.spp.scale0.0.num_batches_tracked, model.spp.scale0.2.weight, model.spp.process1.0.weight, model.spp.process1.0.bias, model.spp.process1.0.running_mean, model.spp.process1.0.running_var, model.spp.process1.0.num_batches_tracked, model.spp.process1.2.weight, model.spp.process2.0.weight, model.spp.process2.0.bias, model.spp.process2.0.running_mean, model.spp.process2.0.running_var, model.spp.process2.0.num_batches_tracked, model.spp.process2.2.weight, model.spp.process3.0.weight, model.spp.process3.0.bias, model.spp.process3.0.running_mean, model.spp.process3.0.running_var, model.spp.process3.0.num_batches_tracked, model.spp.process3.2.weight, model.spp.process4.0.weight, model.spp.process4.0.bias, model.spp.process4.0.running_mean, model.spp.process4.0.running_var, model.spp.process4.0.num_batches_tracked, model.spp.process4.2.weight, model.spp.compression.0.weight, model.spp.compression.0.bias, model.spp.compression.0.running_mean, model.spp.compression.0.running_var, model.spp.compression.0.num_batches_tracked, model.spp.compression.2.weight, model.spp.shortcut.0.weight, model.spp.shortcut.0.bias, model.spp.shortcut.0.running_mean, model.spp.shortcut.0.running_var, model.spp.shortcut.0.num_batches_tracked, model.spp.shortcut.2.weight, model.seghead_extra.bn1.weight, model.seghead_extra.bn1.bias, model.seghead_extra.bn1.running_mean, model.seghead_extra.bn1.running_var, model.seghead_extra.bn1.num_batches_tracked, model.seghead_extra.conv1.weight, model.seghead_extra.bn2.weight, model.seghead_extra.bn2.bias, model.seghead_extra.bn2.running_mean, model.seghead_extra.bn2.running_var, model.seghead_extra.bn2.num_batches_tracked, model.seghead_extra.conv2.weight, model.seghead_extra.conv2.bias, model.final_layer.bn1.weight, model.final_layer.bn1.bias, model.final_layer.bn1.running_mean, model.final_layer.bn1.running_var, model.final_layer.bn1.num_batches_tracked, model.final_layer.conv1.weight, model.final_layer.bn2.weight, model.final_layer.bn2.bias, model.final_layer.bn2.running_mean, model.final_layer.bn2.running_var, model.final_layer.bn2.num_batches_tracked, model.final_layer.conv2.weight, model.final_layer.conv2.bias, loss.criterion.weight


I have update the pretrained weight link, u can retry

**Comment by kaizhong2021** [comment]
> I have update the pretrained weight link, u can retry
I would like to ask you some questions, I want to know how you converted, can I add your WeChat to communicate

File: mmseg/models/backbones/ddrnet.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    Modified from https://github.com/ydhongHIT/DDRNet.
+
+    Args:
+        in_channels
+        channels:
+        ppm_channels
+        align_corners
+        norm_cfg
+        act_cfg
+        init_cfg
```
Comment (xiexinch) [inline mmseg/models/backbones/ddrnet.py:26]: We might complete the docstring.

File: mmseg/models/backbones/ddrnet.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        self.in_channels = in_channels
+        self.ppm_channels = ppm_channels
+
+        self.norm_cfg = norm_cfg
+        self.act_cfg = act_cfg
+        self.align_corners = align_corners
+
+        # stage 0-2
+        self.stem = self._make_stem_layer(in_channels, channels, num_blocks=2)
+        self.relu = nn.ReLU()
```
Comment (xiexinch) [inline mmseg/models/backbones/ddrnet.py:48]: Might support customizing.

File: mmseg/models/decode_heads/ddr_head.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+from mmseg.models.decode_heads.decode_head import BaseDecodeHead
+from mmseg.models.losses import accuracy
+from mmseg.models.utils import resize
+from mmseg.registry import MODELS
+from mmseg.utils import OptConfigType, SampleList
+
+
+@MODELS.register_module()
+class DDRHead(BaseDecodeHead):
+    """Decode head forDDRNet.
```
Comment (xiexinch) [inline mmseg/models/decode_heads/ddr_head.py:17]:
```suggestion
    """Decode head for DDRNet.
```

**Review by xiexinch — COMMENTED** [review]
Hi @whu-pzhang,
Thanks for your contribution!
I think we might add some unit tests for the backbone and head.

**Review by xiexinch — APPROVED** [review]

**Pull request merged**